rust/connlib/clients/shared/src/control.rs

```diff
@@ -46,11 +48,16 @@ fn create_resolver(
         let Ok(Some(dns_servers)) = callbacks.get_system_default_resolvers() else {
             return None;
         };
-        if dns_servers.is_empty() {
+        let mut dns_servers = dns_servers
+            .into_iter()
+            .filter(|ip| ip != &IpAddr::from(DNS_SENTINEL))
+            .peekable();
+        if dns_servers.peek().is_none() {
+            tracing::error!("No system default DNS servers available! Can't initialize resolver. DNS will be broken.");
```

Comment:
should we perhaps just disconnect?

Reply:
Some client presumably won't need or use DNS at all, so maybe this should just be a warning.

Reply:
Ah but this would break DNS system-wide if we're intercepting all queries. Will have to think more about it.

Reply:
Deferring this work to #2939